homeassistant/components/saj/sensor.py

```diff
@@ -238,4 +262,20 @@ def async_update_values(self, unknown_state=False):
     @property
     def unique_id(self):
         """Return a unique identifier for this sensor."""
-        return f"{self._serialnumber}_{self._sensor.name}"
+        return f"{self._inverter.serialnumber}_{self._sensor.name}"
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self._inverter.serialnumber)
+            },
+            "name": "SAJ Solar Inverter",
```

Comment:
Maybe we can set the name to the custom name that was passed along via config flow? (If it's set of course)
Some people have multiple inverters and it would give a better overview in the device overview in my opinion.

Reply:
fixed